span.go

```diff
@@ -155,7 +155,25 @@ func (s *spanImpl) Tracer() opentracing.Tracer {
 }
 
 func (s *spanImpl) Context() opentracing.SpanContext {
-	return s.raw.SpanContext
+	return s.raw.Context
+}
+
+func (s *spanImpl) SetBaggageItem(key, val string) opentracing.Span {
+	s.onBaggage(key, val)
+	if s.trim() {
+		return s
+	}
+
+	s.Lock()
+	defer s.Unlock()
+	s.raw.Context = s.raw.Context.WithBaggageItem(key, val)
```

Comment:
yep. Assuming only a few baggage items are set over the life of a trace, i.e. their number is orders of magnitude fewer than the number of spans in the trace, the copy on write seems like a worthy compromise. 

NB: it's possible to use map implementation based on a linked list, Scala style, which makes copy on write even lighter (0 allocations), without the need to re-allocate the map. I tried it once in our tracer, but abandoned because the implementation of `ForeachBaggageItem` requires iterating through all keys, and doing so via linked list requires keeping a set of visited keys, i.e. another map allocation. However, that issue only occurs if keys are overwritten, so the implementation could've been improved by re-writing the list if keys are rewritten.